test/parallel/test-http-connect.js

```diff
@@ -1,85 +1,90 @@
+  }));
+
+  socket.on('end', common.mustCall(() => {
     socket.end(data);
-  });
-});
-server.listen(0, function() {
-  var req = http.request({
+  }));
+}));
+
+server.listen(0, common.mustCall(function() {
+  const req = http.request({
     port: this.address().port,
     method: 'CONNECT',
     path: 'google.com:443'
   }, common.fail);
 
-  var clientRequestClosed = false;
-  req.on('close', function() {
+  let clientRequestClosed = false;
+  req.on('close', common.mustCall(() => {
     clientRequestClosed = true;
-  });
+  }));
 
-  req.on('connect', function(res, socket, firstBodyChunk) {
+  req.on('connect', common.mustCall((res, socket, firstBodyChunk) => {
     console.error('Client got CONNECT request');
     clientGotConnect = true;
 
     // Make sure this request got removed from the pool.
-    var name = 'localhost:' + server.address().port;
+    const name = 'localhost:' + server.address().port;
     assert(!http.globalAgent.sockets.hasOwnProperty(name));
     assert(!http.globalAgent.requests.hasOwnProperty(name));
 
     // Make sure this socket has detached.
     assert(!socket.ondata);
     assert(!socket.onend);
-    assert.equal(socket.listeners('connect').length, 0);
-    assert.equal(socket.listeners('data').length, 0);
+    assert.strictEqual(socket.listeners('connect').length, 0);
+    assert.strictEqual(socket.listeners('data').length, 0);
 
     // the stream.Duplex onend listener
     // allow 0 here, so that i can run the same test on streams1 impl
     assert(socket.listeners('end').length <= 1);
 
-    assert.equal(socket.listeners('free').length, 0);
-    assert.equal(socket.listeners('close').length, 0);
-    assert.equal(socket.listeners('error').length, 0);
-    assert.equal(socket.listeners('agentRemove').length, 0);
+    assert.strictEqual(socket.listeners('free').length, 0);
+    assert.strictEqual(socket.listeners('close').length, 0);
+    assert.strictEqual(socket.listeners('error').length, 0);
+    assert.strictEqual(socket.listeners('agentRemove').length, 0);
 
-    var data = firstBodyChunk.toString();
-    socket.on('data', function(buf) {
+    let data = firstBodyChunk.toString();
+    socket.on('data', common.mustCall((buf) => {
       data += buf.toString();
-    });
-    socket.on('end', function() {
-      assert.equal(data, 'HeadBody');
+    }));
+
+    socket.on('end', common.mustCall(() => {
+      assert.strictEqual(data, 'HeadBody');
       assert(clientRequestClosed);
       server.close();
-    });
+    }));
+
     socket.write('Body');
     socket.end();
-  });
+  }));
 
   // It is legal for the client to send some data intended for the server
   // before the "200 Connection established" (or any other success or
   // error code) is received.
   req.write('Head');
   req.end();
-});
+}));
 
-process.on('exit', function() {
-  assert.ok(serverGotConnect);
-  assert.ok(clientGotConnect);
+process.on('exit', () => {
+  assert.strictEqual(serverGotConnect, true);
```

Comment:
By using `common.mustCall()`, I think these variables can be eliminated completely, as can this `exit` event handler.